lib/ftp.js

```diff
@@ -67,13 +67,16 @@ class Client {
             port,
             family: this.ftp.ipFamily
         }, () => this.ftp.log(`Connected to ${describeAddress(this.ftp.socket)}`));
-        return this.ftp.handle(undefined, (res, task) => {
-            if (positiveCompletion(res.code)) {
+        return this.ftp.handle(undefined, (err, res, task) => {
+            if (err) {
+                task.reject(err);
+            }
+            else if (positiveCompletion(res.code)) {
```

Comment:
This is a new pattern for tasks: Use some kind of guard-clause for errors. Tasks know about errors mainly to have the chance to do some cleanup, see upload and download. Most of the time, though, tasks just immediately reject.